src/util/cast.js

```diff
@@ -100,6 +100,12 @@ Cast.toRgbColorObject = function (value) {
 Cast.compare = function (v1, v2) {
     var n1 = Number(v1);
     var n2 = Number(v2);
+    if (n1 === 0 && (v1 == null || typeof v1 === 'string' && v1.trim().length === 0)) {
```

Comment:
This string of comparators should probably be moved out into it's own `util/cast` method for the sake of readability, reusability, and testing. Perhaps we could call it simply `isEmpty` and then amend the unit test suite for `compare` and add test coverage for `isEmpty`?